features/gba-dev/europees-kiesrecht-gba.feature

```diff
@@ -0,0 +1,62 @@
+
+  Scenario: Geen Europees kiesrecht opgegevens, geen Europees kiesrecht (31.10) en geen einddatum uitsluiting Europees kiesrecht (31.30)
+    Gegeven de persoon met burgerservicenummer '000000103' heeft de volgende gegevens
+    | geslachtsnaam (02.40) |
+    | Janssen               |
+    Als gba personen wordt gezocht met de volgende parameters
+    | naam                | waarde                          |
+    | type                | RaadpleegMetBurgerservicenummer |
+    | burgerservicenummer | 000000103                       |
+    | fields              | europeesKiesrecht            |
+    Dan heeft de response een persoon zonder gegevens
+
+  Abstract Scenario: Europees kiesrecht zonder einddatum : <omschrijving>
+    Gegeven de persoon met burgerservicenummer '000000115' heeft de volgende 'inschrijving' gegevens
+    | Europees kiesrecht (31.10) |
+    | <Europees kiesrecht>       |
+    Als gba personen wordt gezocht met de volgende parameters
+    | naam                | waarde                          |
+    | type                | RaadpleegMetBurgerservicenummer |
+    | burgerservicenummer | 000000115                       |
+    | fields              | europeesKiesrecht               |
+    Dan heeft de response een persoon met de volgende 'europeesKiesrecht' gegevens
+    | naam                     | waarde                    |
+    | aanduiding.code          | <aanduiding.code>         |
+    | aanduiding.omschrijving  | <aanduiding.omschrijving> |
+
+    Voorbeelden:
+   | omschrijving                       | Europees kiesrecht | aanduiding.code | aanduiding.omschrijving |
+   | uitgesloten zonder einddatum       | 1                  | 1               | persoon is uitgesloten  |
+   | persoon ontvangt oproep            | 2                  | 2               | persoon ontvangt oproep |
+
+
+  Abstract Scenario: Europees kiesrecht <omschrijving>
+    Gegeven de persoon met burgerservicenummer '000000127' heeft de volgende 'inschrijving' gegevens
+    | Europees kiesrecht (31.10) | einddatum uitsluiting Europees kiesrecht (31.30) |
+    | <Europees kiesrecht>       | <einddatum uitsluiting>                          |
+    Als gba personen wordt gezocht met de volgende parameters
+    | naam                | waarde                                |
+    | type                | RaadpleegMetBurgerservicenummer       |
+    | burgerservicenummer | 000000127                             |
+    | fields              | burgerservicenummer,europeesKiesrecht |
+    Dan heeft de response een persoon met de volgende gegevens
+    | naam                                       | waarde                    |
+    | burgerservicenummer                        | 000000127                 |
+    | europeesKiesrecht.aanduiding.code          | <aanduiding.code>         |
+    | europeesKiesrecht.aanduiding.omschrijving  | <aanduiding.omschrijving> |
+    | europeesKiesrecht.einddatumUitsluiting     | <einddatumUitsluiting>    |
+
+    Voorbeelden:
```

Comment:
Ik begrijp in deze tabel de waarden van de 3e en 6e kolom niet.
Volgens het LO GBA kan het veld 'einddatum uitsluiting Europees kiesrecht (31.30)' de volgende mogelijke waarden hebben:
* jjjjmmdd
* jjjjmm00
* jjjj0000
* 00000000

Een waarde als bijv. 'volgend jaar' is volgens mij dus niet correct.
De GBA API specificeert vervolgens op haar beurt voor de property 'europeesKiesrecht.einddatumUitsluiting' het type 'GbaDatum'. Dat moet dus ook een datum zijn en ook hier geldt dat een waarde als 'volgend jaar' volgens mij niet correct is.

Is er in de e2e automation code die waardes als 'volgend jaar' en 'gisteren' omzet naar relevante datums?

Reply:
> Ik begrijp in deze tabel de waarden van de 3e en 6e kolom niet. Volgens het LO GBA kan het veld 'einddatum uitsluiting Europees kiesrecht (31.30)' de volgende mogelijke waarden hebben:
> 
> * jjjjmmdd
> * jjjjmm00
> * jjjj0000
> * 00000000
> 
> Een waarde als bijv. 'volgend jaar' is volgens mij dus niet correct. De GBA API specificeert vervolgens op haar beurt voor de property 'europeesKiesrecht.einddatumUitsluiting' het type 'GbaDatum'. Dat moet dus ook een datum zijn en ook hier geldt dat een waarde als 'volgend jaar' volgens mij niet correct is.
> 
> Is er in de e2e automation code die waardes als 'volgend jaar' en 'gisteren' omzet naar relevante datums?

Ja, in de automation code wordt waardes als 'volgend jaar' en 'gisteren' omgezet naar relevante datums. Voor ondersteunde waardes, zie de uitgecommentarieerde code onderaan in de file: https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/blob/develop/features/step_definitions/calcDate.js